packages/react-dom-bindings/src/client/ReactFiberConfigDOM.js

```diff
@@ -1311,19 +1311,18 @@ export function registerSuspenseInstanceRetry(
 ) {
   const ownerDocument = instance.ownerDocument;
   if (ownerDocument.readyState !== DOCUMENT_READY_STATE_COMPLETE) {
-    ownerDocument.addEventListener(
-      'DOMContentLoaded',
-      () => {
-        if (instance.data === SUSPENSE_PENDING_START_DATA) {
-          callback();
-        }
-      },
-      {
-        once: true,
-      },
-    );
+    const listener = () => {
+      if (instance.data === SUSPENSE_PENDING_START_DATA) {
+        callback();
+      }
+      ownerDocument.removeEventListener('DOMContentLoaded', listener);
+    };
+    ownerDocument.addEventListener('DOMContentLoaded', listener);
+    instance._reactRetry = callback;
+  } else {
```

Comment:
What happens if the boundary has been resolved before the retry is registered and before the document is resolved?

There’s a race condition there.

I think it should be sufficient to check the status of the comment node again here.

Reply:
yeah, that's right